pkg/validation/variables/utils.go

```diff
@@ -0,0 +1,44 @@
+package variables
+
+import (
+	"strings"
+
+	"github.com/sirupsen/logrus"
+)
+
+func init() {
+	// initialize the logrus time formatter
+	formatter := &logrus.TextFormatter{
+		FullTimestamp: true,
+	}
+	logrus.SetFormatter(formatter)
+}
+
+// logInvalidVariables logs a warning for all the invalid keys belong to a devfile object.
+//  We do not want to err out but only log a warning in case if variable key is not resolved.
+func logInvalidVariables(objectType, objectName string, invalidKeys []string) {
+	logrus.Warnf("Devfile %s %s has invalid variable references - %s", objectType, objectName, strings.Join(invalidKeys, ","))
```

Comment:
I don't think it's acceptable to log in a library function -- if I call this from the controller and there's an issue, is it going to print logs in its own format? Library functions should return errors (or a 'warning' error type) that can be printed however the consumer deems appropriate.

Reply:
Updated, used a warning struct as discussed on Slack :)